core/src/data_sources/data_source.rs

```diff
@@ -594,23 +614,87 @@ impl DataSource {
             utils::now() - now
         ));
 
+        let now = utils::now();
+        let mut embeddings: HashMap<String, EmbedderVector> = HashMap::new();
+        let mut page_offset: Option<PointId> = None;
+        loop {
+            let scroll_results = qdrant_client
+                .scroll(&ScrollPoints {
+                    collection_name: self.qdrant_collection(),
+                    with_vectors: Some(true.into()),
+                    limit: Some(1024),
```

Comment:
The goal of power of 2 is quite arbitrary, it's just that it might help qdrant optimize in some cases vs page sizes etc...